pkg/collector/corechecks/net/wlan/wlan_darwin.go

```diff
@@ -20,6 +20,38 @@ import (
 	"unsafe"
 )
 
+// phyMode represents the PHY mode of the WiFi connection.
+type phyMode int
+
+// See: https://developer.apple.com/documentation/corewlan/cwphymode?language=objc
+const (
+	phyModeNone phyMode = iota // No PHY mode.
+	phyMode11a                 // IEEE 802.11a PHY.
+	phyMode11b                 // IEEE 802.11b PHY.
+	phyMode11g                 // IEEE 802.11g PHY.
+	phyMode11n                 // IEEE 802.11n PHY.
+	phyMode11ac                // IEEE 802.11ac PHY.
+)
+
+func formatPhy(phy phyMode) string {
```

Comment:
```suggestion
func (phyMode p) String() string {
```
Let's define the `String()` method instead to be a [Stringer](https://go.dev/tour/methods/17) Interface.

Reply:
Sure, if it is more idiomatic (in fact CoPilot suggested just that). I just did similar things for windows code. I did not feel that it worth to objectivezing half a dozen of enumerations. But I will do your suggestion, thank you.

Reply:
Also I am replacing "unknown" on "" because in wlan.go (and tests) I saw substitution of "" on "unknown"

Reply:
Yes, that's the idiomatic way of doing string enums in go, example: https://gobyexample.com/enums